cmd/pmm-managed/main.go

```diff
@@ -237,6 +242,65 @@ func runDebugServer(ctx context.Context) {
 	cancel()
 }
 
+func runTelemetryService(ctx context.Context) {
+	l := logrus.WithField("component", "TELEMETRY")
+
+	disabled := strings.TrimSpace(strings.ToLower(os.Getenv("DISABLE_TELEMETRY")))
+	if disabled == "true" || disabled == "1" || disabled == "yes" {
+		l.Infof("Telemetry is disabled by DISABLE_TELEMETRY env var")
+		return
+	}
+
+	cfg := telemetry.DefaultConfig()
+	if uuid, err := getTelemetryUUID(*consulAddress, cfg.UUID); err != nil {
+		l.Warnf("cannot get/set telemetry UUID in consul: %s", err)
+	} else {
+		cfg.UUID = uuid
+	}
+
+	// Using this env var for compatibility with the Toolkit
+	if telemetryEnvURL := os.Getenv("PERCONA_VERSION_CHECK_URL"); telemetryEnvURL != "" {
+		l.Infof("PERCONA_VERSION_CHECK_URL env var is set")
+		l.Infof("Using %s as the telemetry endpoint", telemetryEnvURL)
+		cfg.URL = telemetryEnvURL
+	}
+
+	svc, err := telemetry.NewService(cfg)
+	if err != nil {
+		l.Warnf("Cannot start telemetry: %s", err.Error())
+		return
+	}
+
+	l.Infof("Telemetry is enabled. Send data interval = %v", cfg.Interval)
+	svc.Start()
+
+	<-ctx.Done()
+	svc.Stop()
+}
+
+func getTelemetryUUID(consulAddress, defaultUUID string) (string, error) {
+	client, err := consulapi.NewClient(&consulapi.Config{Address: consulAddress})
```

Comment:
I added consul service in `master`. Accept it as a parameter and use it, so we don't have several clients/connections in various places.